src/components/motor/MotorController.cpp

```diff
@@ -13,19 +14,43 @@ MotorController::MotorController(Controllers::Settings& settingsController) : se
 void MotorController::Init() {
   nrf_gpio_cfg_output(pinMotor);
   nrf_gpio_pin_set(pinMotor);
-  app_timer_create(&vibTimer, APP_TIMER_MODE_SINGLE_SHOT, vibrate);
+  app_timer_init();
+  
+  app_timer_create(&shortVibTimer, APP_TIMER_MODE_SINGLE_SHOT, vibrate);
+  app_timer_create(&longVibTimer, APP_TIMER_MODE_REPEATED, vibrate);
+  isBusy = false;
 }
 
-void MotorController::SetDuration(uint8_t motorDuration) {
+void MotorController::runForDuration(uint8_t motorDuration) {
 
-  if (settingsController.GetVibrationStatus() == Controllers::Settings::Vibration::OFF)
+  if (settingsController.GetVibrationStatus() == Controllers::Settings::Vibration::OFF || isBusy)
     return;
 
   nrf_gpio_pin_clear(pinMotor);
   /* Start timer for motorDuration miliseconds and timer triggers vibrate() when it finishes*/
-  app_timer_start(vibTimer, APP_TIMER_TICKS(motorDuration), NULL);
+  app_timer_start(shortVibTimer, APP_TIMER_TICKS(motorDuration), NULL);
 }
 
-void MotorController::vibrate(void* p_context) {
+void MotorController::startRunning(uint8_t motorDuration) {
```

Comment:
uint8_t isn't big enough

Reply:
Or maybe we don't need this parameter at all.

Reply:
It defines the length of a the individual vibration and wait times the continuous pattern is made up of. You're right that using uint8_t limits that lenght to 256ms. Ill switch to 16 bit, but its really not that big of a deal (the only place this is currently used, the call notification sets it to 10ms). Removing this would mean hard-coding the "period" of the pattern which would mean applications cant vary their patterns.

Reply:
Exactly, however I think it might be better for consistency to not let applications change it. And "hard coding" isn't always a bad thing if the alternative is to re define the same thing in many places.

In one of the earlier commits, there was this line, which overflowed the parameter. I don't know if you changed this later, as currently this line doesn't exist in this PR.
`motorController.startRunning(500);`

Reply:
Hardcoding this parameter is a possibility, indeed, mostly if we only use 1 single value.
If you want to future proof the class, the parameter is nice too. IMHO, the best solution for the type is not uint8_t or uint16_t or uint32_t but a stronger type that clearly represents a duration : `std::chrono::millisecond` or a custom type `Duration`, for example.

Reply:
Agree with using `std::chrono::milliseconds`, it's large enough and eliminate any ambiguity as to what the expected units of `motorDuration` are without needing to dig into the implementation.